src/extensions/default/AutoUpdate/UpdateInfoBar.js

```diff
@@ -0,0 +1,167 @@
+     * @param   {object}   msgObj - json object containing message information to be displayed
+     * @returns {object} - the generated json object
+     */
+    function generateJsonForMustache(msgObj) {
+        var msgJsonObj = {};
+        if (msgObj.type) {
+            msgJsonObj.type = "'" + msgObj.type + "'";
+        }
+        msgJsonObj.title = msgObj.title;
+        msgJsonObj.description = msgObj.description;
+        if (msgObj.needButtons) {
+            msgJsonObj.buttons = [{
+                "id": "restart",
+                "value": Strings.RESTART_BUTTON,
+                "tIndex": "'0'"
+            }, {
+                "id": "later",
+                "value": Strings.LATER_BUTTON,
+                "tIndex": "'0'"
+            }];
+            msgJsonObj.needButtons = msgObj.needButtons;
+        }
+        return msgJsonObj;
+    }
+
+    /**
+     * Removes and cleans up the update bar from DOM
+     */
+    function cleanUpdateBar() {
+        var $updateBar = $('#update-bar');
+        if($updateBar.length > 0) {
+            $updateBar.remove();
+        }
+        $(window.document).off("keydown.AutoUpdate");
+    }
+
+    /**
+     * Displays the Update Bar UI
+     * @param   {object} msgObj - json object containing message info to be displayed
+     *
+     */
+    function showUpdateBar(msgObj) {
+        var jsonToMustache = generateJsonForMustache(msgObj),
+            $updateBarElement = $(Mustache.render(UpdateBarHtml, jsonToMustache));
+
+        cleanUpdateBar(); //Remove an already existing update bar, if any
+        $updateBarElement.prependTo(".content");
+
+        var $updateBar = $('#update-bar'),
+            $updateContent = $updateBar.find('#update-content'),
+            $contentContainer = $updateBar.find('#content-container'),
+            $heading = $updateBar.find('#heading'),
+            $description = $updateBar.find('#description'),
+            $restart = $updateBar.find('#update-btn-restart'),
+            $later = $updateBar.find('#update-btn-later'),
+            $closeIcon = $updateBar.find('#close-icon');
+
+        if($updateContent.length > 0) {
+            if ($updateContent[0].scrollWidth > $updateContent.innerWidth()) {
+            //Text has over-flown, show the update content as tooltip message
+                if($contentContainer.length > 0 &&
+                   $heading.length > 0 &&
+                   $description.length > 0) {
+                    $contentContainer.attr("title", $heading.text() + $description.text());
+                }
+            }
+        }
+
+        //Event handlers on the Update Bar
+
+        if($restart.length > 0) {
+            $restart.click(function () {
+                cleanUpdateBar();
+                MainViewManager.trigger(exports.RESTART_BTN_CLICKED);
+            });
+
+            $restart.keyup(function (e) {
+                if (e.which === 32) { //32 is the keycode for space key
```

Comment:
use a named variable for better readability.

Reply:
Addressed in  537430b